provider/src/main/java/ru/yandex/yamblz/euv/provider/ArtistProvider.java

```diff
@@ -0,0 +1,75 @@
+package ru.yandex.yamblz.euv.provider;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.content.UriMatcher;
+import android.database.Cursor;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+
+import ru.yandex.yamblz.euv.provider.db.DbBackend;
+import ru.yandex.yamblz.euv.shared.contract.ProviderContract;
+import timber.log.Timber;
+
+/**
+ * TODO: In additional to inserting initial data into db from json asset,
+ * TODO: perform sync operations via {@link android.content.AbstractThreadedSyncAdapter}
+ */
+public class ArtistProvider extends ContentProvider {
+    private static final UriMatcher uriMatcher = new UriMatcher(UriMatcher.NO_MATCH);
+    private static final int ALL_ARTISTS = 0;
+    private static final int SINGLE_ARTIST = 1;
+
+    static {
+        uriMatcher.addURI(ProviderContract.AUTHORITY, ProviderContract.ARTISTS_PATH, ALL_ARTISTS);
+        uriMatcher.addURI(ProviderContract.AUTHORITY, ProviderContract.ARTISTS_PATH + "/#", SINGLE_ARTIST);
+    }
+
+    private DbBackend dbBackend;
+
+    @Override
+    public boolean onCreate() {
+        dbBackend = new DbBackend(getContext());
```

Comment:
Из документации к методу ContentProvider#onCreate():
```
     * This method is called for all registered content providers on the
     * application main thread at application launch time.  It must not perform
     * lengthy operations, or application startup will be delayed.
...
     * If you use SQLiteOpenHelper, make sure to avoid calling
     * SQLiteOpenHelper#getReadableDatabase() or
     * SQLiteOpenHelper#getWritableDatabase()
     * from this method.  (Instead, override
     * SQLiteOpenHelper#onOpen() to initialize the
     * database when it is first opened.)
```
Исходя из этого, наполнять БД начальными данными в момент её создания (которое, при этом, отличается от момента инициализации CP) я посчитал хорошей идеей.